components/dashboards/org.wso2.carbon.dashboards.core/src/main/java/org/wso2/carbon/dashboards/core/internal/DashboardMetadataProviderImpl.java

```diff
@@ -454,4 +454,26 @@ public boolean isCreator(String username) throws DashboardException {
         }
         return isCreator;
     }
+
+    @Override
+    public boolean isWidgetCreator(String username) throws DashboardException {
+        List<org.wso2.carbon.analytics.idp.client.core.models.Role> userRoles;
+        boolean isWidgetCreator = false;
+        try {
+            userRoles = identityClient.getUserRoles(username);
+        } catch (IdPClientException e) {
+            throw new DashboardException("Unable to get roles for the username.");
```

Comment:
e is missing throwing new DashboardException

Reply:
Fixed in dfb28946c56f870908f491b93539f7e8ac4b0c06